dev/tools/localization/gen_l10n.dart

```diff
@@ -473,13 +470,109 @@ Future<void> main(List<String> arguments) async {
     'pub',
     'run',
     'intl_translation:generate_from_arb',
-    '--output-dir=${l10nDirectory.path}',
+    '--output-dir=${localizationsGenerator.l10nDirectory.path}',
     '--no-use-deferred-loading',
-    outputFile.path,
+    localizationsGenerator.outputFile.path,
     ...arbFilenames,
   ]);
   if (generateFromArbResult.exitCode != 0) {
     stderr.write(generateFromArbResult.stderr);
     exit(1);
   }
 }
+
+/// The localizations generation class used to generate the localizations
+/// classes, as well as all pertinent Dart files required to internationalize a
+/// Flutter application.
+class LocalizationsGenerator {
+  /// Creates an instance of the localizations generator class.
+  ///
+  /// It takes in a [FileSystem] representation that the class will act upon.
+  LocalizationsGenerator(this._fs);
+
+  final file.FileSystem _fs;
+
+  /// The reference to the project's l10n directory.
+  ///
+  /// It is assumed that all input files (ie. [templateArbFile], arb files
+  /// for translated messages) and output files (ie. The localizations
+  /// [outputFile], `messages_<locale>.dart` and `messages_all.dart`)
+  /// will reside here.
+  Directory l10nDirectory;
+
+  /// The reference to the template arb file.
+  ///
+  /// The messages and its metadata will be used to generate the
+  /// `messages_<locale>.dart` and `messages_all.dart` files.
+  File templateArbFile;
+
+  /// The reference to the Dart file will contain the localizations and
+  /// localizations delegate classes.
+  File outputFile;
+
+  /// The class name to be used for the localizations class in [outputFile].
+  ///
+  /// For example, if 'AppLocalizations' is passed in, the AppLocalizations
+  /// class will be created with a [LocalizationsDelegate] named
+  /// _AppLocalizationsDelegate.
+  String className;
+
+  /// Sets the reference [Directory] for [l10nDirectory].
+  void setL10nDirectory(String arbPathString) {
```

Comment:
I would create a single `initialize` method which takes all of this data as named arguments. Then later on, you can throw a StateError if anything is accessed before this method is called